src/array.cr

```diff
@@ -607,20 +639,18 @@ class Array(T)
   # a[6..10]? # => nil
   # a[6..]?   # => nil
   # ```
-  def []?(range : Range)
+  def []?(range : Range) : self | Nil
     self[*Indexable.range_to_index_and_count(range, size) || return nil]?
   end
 
   # Returns count or less (if there aren't enough) elements starting at the
   # given start index.
```

Comment:
I think it would be a good idea to call this "begin" instead of "start", especially since this method accepts a `Range`?

Reply:
The equivalent method receiving two ints has `start` as first argument, so I think it's best to stay with that same term.

Ideally, yeah maybe it could all be `begin`, but I don't think it's worth changing that.